src/ValueObject/EncryptedColumn.php

```diff
@@ -43,4 +63,25 @@ public function getData()
     {
         return $this->data;
     }
+
+    /**
+     * @return string
+     */
+    public function getEncryptor(): string
+    {
+        return $this->encryptor;
+    }
+
+    /**
+     * @return string
+     */
+    public function getSerializer(): string
+    {
+        return $this->serializer;
+    }
+
+    public function needsReencryption($encryptor, $serializer)
+    {
+        return $encryptor === $this->encryptor && $serializer === $this->serializer;
```

Comment:
As mentioned above, an encryptor may be parametrized. An identifier may not be able to contain all information about the parameters.